### ISSUE TITLE
Notification email is sent to "Responsible" of Vendor Bills

### ISSUE THREAD

**Issue by TimLai125** [odoo/odoo#20665]
Impacted versions:
10.0 and probably for 11.0 as well.

What are the steps to reproduce your issue? 
1. Create a Vendor Bills. 
2. Select any "Vendor". 
3. Select a "Responsible" who is not the current login-ed user. 
4. Validate the Vendor Bill. 

What is the current behavior that you observe? 
Even "Email and Notification" of the "Responsible" is set to "Never", the email notification still sends to the email of the "Responsible". 

What would be your expected behavior in this case? 
No email will be sent to the customers / users with email notification setting is set to "Never". 

Video/Screenshot link (optional):
https://youtu.be/qVs8Pkzhhqo

**Comment by TimLai125** [comment]
There is similar issues on the sales orders and customer invoices as well, selecting a different salesperson that is having "Never" as the "Email and Notification" setting, notification email is still sent to the salesperson once the confirmation of sales orders and validation of invoices.

**Comment by Yenthe666** [comment]
@jco-odoo I don't really think this is desired behaviour, is it?

**Comment by jco-odoo** [comment]
@lap-odoo @tde-banana-odoo Is not the responsible to be in the chatter as any other user?

**Comment by pedrobaeza** [comment]
In the case of vendor bills, I think there's other issue: members of the sale team of the responsible are also added as followers. If you consider, I can put a separate issue for this.

**Comment by tde-banana-odoo** [comment]
Hello,

It is the intended behavior as far as I know. Responsible of records should always be warned they are responsible independently of the "notification email" parameter. This is something that has been implemented in v10 and it still present in v11. It will probably change for v12, but I cannot tell you how as nothing is completely specified.

**Comment by pedrobaeza** [comment]
@tde-banana-odoo what about the team as followers behavior? Have you tried?

**Comment by TimLai125** [comment]
@tde-banana-odoo I think what is being expected by user is the "Email and Notification" settings should have a higher priority in controlling these kind of behaviors.

May I know how Odoo classify notification that it must send to users? As the settings seem to be vague if it does not cover all the notification emails.

**Comment by rmouthier** [comment]
@tde-banana-odoo do you have any update on this?

**Comment by tde-banana-odoo** [comment]
Hello,

Actually the settings does cover all notifications linked to chatter. Assignation is a pure email sent to the responsible and short-circuits the whole process. If you look carefully once the email is sent there should be no entry in the chatter.

In v12 assignation for responsibles will be classic chatter post. A note with the responsible as recipient, meaning he will either receive an Inbox notification, either an email. This makes the behavior more consistent and will probably simplify the understanding of the mechanism.

Thanks,

**Comment by rmouthier** [comment]
@tde-banana-odoo thx for your input. Much appreciated 👍

**Comment by TimLai125** [comment]
@tde-banana-odoo 
Thank you for your explanation. For my understanding, might I know why the behavior is different from "Vendor Bills" and "Customer Invoices"? Since for "Vendor Bills", the assignation is not linked to chatter while "Customer Invoice" does. 

Furthermore, is there any method to prevent those "Assignation" emails from sending to the "Responsible"?

**Comment by TimLai125** [comment]
@tde-banana-odoo May I have your reply please?

**Comment by tde-banana-odoo** [comment]
Hi @TimLai125 

Assignation email is the same for all models. If there is a responsible field, an email is sent whatever the user preferences. In 11.0 this is the intended behavior and it cannot be prevented. It is about to change in master, meaning for v12 it will be a standard notification following user choice.

Current behavior is not considered as a bug. The feature is not very smart and that is why it will change in a near future. However in stable we don't change behavior and it will stay as it is.

Concerning invoices, it should behave the same. I am not aware of differences; maybe somehow the email was not sent correctly due to email server issues. When the email is not sent it appears in the chatter due to some technical implementation details. Once the email is sent it is not displayed anymore in the chatter.

I hope it answers your concern. I will try to keep you informed of changes in the behavior once it lands in master if you want so.

Best regards,

**Comment by TimLai125** [comment]
@tde-banana-odoo Thank you for the reply. I believe it will be better for Odoo to clarify the behavior in the  "Email and Notification" settings, perhaps a help text.

**Comment by tde-banana-odoo** [comment]
Well technically it already tells your it handle Chatter notifications. Assignation email is a real email and not a Chatter notification. But I agree it is not clear enough and require a deep understanding of the chatter and mailing mechanism. That is why we are changing the behavior to simplify it as a lot of people have the same question.
